### ISSUE TITLE
Publish for netstandardapp1.5 no longer works

### ISSUE THREAD

**Issue by moozzyk** [dotnet/cli#2482]
## Steps to reproduce
1. `dotnet.exe publish -o C:\temp\reprobreaking1 --framework netstandardapp1.5`
2. Run the published application

## Expected  behavior
Application runs

## Actual behavior
```
PS C:\temp\reprobreaking1> .\TestSites.exe
Invalid .deps file
```

The .deps file actually does not exist 

## Environment data
`dotnet --info` output:

1.0.0-rc2-002374, 1.0.0-rc2-002376

##Trace

```
PS C:\temp\reprobreaking1> .\TestSites.exe
Tracing enabled
--- Invoked host main = {
C:\temp\reprobreaking1\TestSites.exe
}
Resolved fxr [C:\temp\reprobreaking1\hostfxr.dll]...
Loaded library from C:\temp\reprobreaking1\hostfxr.dll
Tracing enabled
Exists C:\temp\reprobreaking1\TestSites.dll
Checking if CoreCLR path exists=[C:\temp\reprobreaking1\coreclr.dll]
Detecting mode... CoreCLR present in own dir [C:\temp\reprobreaking1] and checking if [TestSites.deps.json] file present=[1]
Host operating from standalone app dir C:\temp\reprobreaking1
Loaded library from C:\temp\reprobreaking1\hostpolicy.dll
Tracing enabled
Invalid .deps file
```

## Project.json

```
{
  "version": "1.0.0-*",
  "compilationOptions": {
    "emitEntryPoint": true
  },
  "content": [
    "web.config"
  ],
  "dependencies": {
    "Microsoft.AspNetCore.Server.IISIntegration": "1.0.0-*",
    "Microsoft.AspNetCore.Server.Kestrel": "1.0.0-*",
    "Microsoft.AspNetCore.WebUtilities": "1.0.0-*",
    "Microsoft.Extensions.Configuration.EnvironmentVariables": "1.0.0-*",
    "Microsoft.Extensions.Configuration.Json": "1.0.0-*",
    "Microsoft.Extensions.Logging.Console": "1.0.0-*",
    "xunit": "2.1.0"
  },
  "frameworks": {
    "netstandardapp1.5": {
      "imports": [
        "portable-net451+win8",
        "dnxcore50"
      ],
      "dependencies": {
        "System.Net.Primitives": "4.0.11-*",
        "dotnet-test-xunit": "1.0.0-dev-*",
        "NETStandard.Library": "1.5.0-*",
        "System.Diagnostics.Process": "4.1.0-*"
      }
    },
    "net451": {}
  }
}

```

**Comment by eerhardt** [comment]
I think this is fixed with #2473.  Going to test out locally to make sure.

**Comment by brthor** [comment]
This scenario is currently tested in the repo, I'm surprised to see a failure.  

If you do a dotnet build do you see a deps.json in the output?

**Comment by moozzyk** [comment]
Yes. I can `dotnet run` in the folder and the deps.json file is being created.

**Comment by piotrpMSFT** [comment]
Projects targetting NetStandardApp need to be ported to NetCoreApp. NetStandardApp is not going to be supported by CLI 1.0.0 RC2. 

/cc @Petermarcu @richlander

**Comment by davidfowl** [comment]
```
> PS C:\temp\reprobreaking1> .\TestSites.exe
Invalid .deps file
```

@piotrpMSFT The error message is a bit terse no? Shouldn't it say something helpful? It took 2 devs > 30 minutes to figure this out...

**Comment by moozzyk** [comment]
This is not this. It turned out that the app had a dependency on xunit which brought old NetCore.App and this is where hostpolicy.dll and hostfxr were copied from and there was a mismatch. How anyone can figure this out if @pakrym  is not around?

**Comment by moozzyk** [comment]
(It's no longer blocking though)

**Comment by danroth27** [comment]
Sounds like @pakrym has a solid consulting career ahead of him :smile:

**Comment by moozzyk** [comment]
Looks like this may affect any standalone application if it depends on a package that brings an older/incompatible NETCoreApp package.

**Comment by GuardRex** [comment]
> Invalid .deps file

/cc @ciel ... look familiar?

**Comment by mjrousos** [comment]
I'm hitting this as well with a simple 'Hello World' app.

project.json:
```json
{
  "version": "1.0.0-*",
  "compilationOptions": {
    "emitEntryPoint": true
  },
  "dependencies": {
    "Microsoft.NETCore.App": "1.0.0-rc2-24018"
  },
  "frameworks": {
    "netcoreapp1.0": {}
  },
  "runtimes": {
    "win10-x64":{}
  }
}
```

NuGet.config:
```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="dotnet-core" value="https://dotnet.myget.org/F/dotnet-core/api/v3/index.json" />
    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
  </packageSources>
</configuration> 
```

Builds/runs fine via `dotnet` commands. Running from the published directory yields the 'Invalid .deps file' error.

**Comment by eerhardt** [comment]
@mjrousos - you are using the wrong version of Microsoft.NETCore.App.  This package doesn't have the same versioning scheme as the rest of the CoreFX packages.

```
"Microsoft.NETCore.App": "1.0.0-rc2-24018"
```

Instead you want:

```
"Microsoft.NETCore.App": "1.0.0-rc2-3002416"
```

A way to tell the difference is that the bad/wrong packages have rc2-2xxxxx.  The correct packages have rc2-3xxxxxx.  If you want the latest version, you should just use rc2-*.

**Comment by mjrousos** [comment]
Ah, I see. Going forward what's the best way for me to know what a recent dev build of Microsoft.NETCore.App is? For most assemblies, I just check [myget](https://dotnet.myget.org/gallery/dotnet-core), but it shows an incorrect version for Microsoft.NETCore.App (which is how I ended up in this state in the first place :)).

**Comment by eerhardt** [comment]
If you look at the listing of all versions in the myget feed, you can see all the ones with "rc2" in them.  The highest versions (i.e. the ones with rc2-3xxxxxx) will be the most recent.

**Comment by GuardRex** [comment]
You can cap the rc2 ones if you use the CI release feed (as shown in the
cli samples repo), which is being used for rc2 stabilization prior to rc2
release. If you use the CI dev feed, you float into rc3.

It's probably best to follow the format of cli samples right now for shared
framework apps with mods if you wish to embrace the standalone model.
On Apr 18, 2016 11:56 AM, "Eric Erhardt" <notifications@github.com> wrote:

If you look at the listing of all versions in the myget feed, you can see
all the ones with "rc2" in them. The highest versions (i.e. the ones with
rc2-3xxxxxx) will be the most recent.

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub
<https://github.com/dotnet/cli/issues/2482#issuecomment-211472246>

**Comment by mjrousos** [comment]
Got it. Thanks, @eerhardt and @GuardRex!
